src/engine/sidechain/shoutconnection.cpp

```diff
@@ -776,22 +779,22 @@ bool ShoutConnection::metaDataHasChanged() {
 
     m_iMetaDataLife = 0;
 
-    pTrack = PlayerInfo::instance().getCurrentPlayingTrack();
-    if (!pTrack) {
+    TrackPointer pPlayingTrack = PlayerInfo::instance().getCurrentPlayingTrack();
+    if (!pPlayingTrack) {
         return false;
     }
 
-    if (m_pMetaData) {
-        if (!pTrack->getId().isValid() || !m_pMetaData->getId().isValid()) {
-            if ((pTrack->getArtist() == m_pMetaData->getArtist()) &&
-                (pTrack->getTitle() == m_pMetaData->getArtist())) {
+    if (m_pMetaDataTrack) {
+        if (!pPlayingTrack->getId().isValid() || !m_pMetaDataTrack->getId().isValid()) {
+            if ((pPlayingTrack->getArtist() == m_pMetaDataTrack->getArtist()) &&
+                    (pPlayingTrack->getTitle() == m_pMetaDataTrack->getArtist())) {
                 return false;
             }
-        } else if (pTrack->getId() == m_pMetaData->getId()) {
+        } else if (pPlayingTrack->getId() == m_pMetaDataTrack->getId()) {
             return false;
         }
     }
```

Comment:
This nested if feels more complicated to understand that it should, could we simplify to something like

```
    if (m_pMetaDataTrack && 
        (!pPlayingTrack->getId().isValid() || !m_pMetaDataTrack->getId().isValid()) &&
        pPlayingTrack->getArtist() == m_pMetaDataTrack->getArtist() &&
        pPlayingTrack->getTitle() == m_pMetaDataTrack->getArtist()) {
        return false;
    } else if (m_pMetaDataTrack && pPlayingTrack->getId() == m_pMetaDataTrack->getId()) {
        return false;
    }
```